utils/workflows/files.go

```diff
@@ -88,13 +89,63 @@ func ListFiles(ctx workflow.Context, path paths.Path) (paths.Files, error) {
 }
 
 func UnmarshalXMLFile[T any](ctx workflow.Context, file paths.Path) (*T, error) {
-	var r T
 	res, err := ReadFile(ctx, file)
 	if err != nil {
 		return nil, err
 	}
-	err = xml.Unmarshal(res, &r)
-	return &r, err
+
+	unmarsallResult := workflow.SideEffect(ctx, func(ctx workflow.Context) any {
+		var r T
+		err := xml.Unmarshal(res, &r)
+		if err != nil {
+			panic(err)
+		}
+		return r
+	})
+
+	var data T
+	unmarsallResult.Get(data)
+
+	return &data, nil
+}
+
+func SafeMarshallXml(ctx workflow.Context, data any) ([]byte, error) {
+	var res []byte
+	err := workflow.SideEffect(ctx, func(ctx workflow.Context) any {
```

Comment:
feel `safe` is misdescriptive.
Maybe have a package called `sideeffect`: sideeffect.marshalXml(), sideeffect.parsePath(), etc.
Or wfutils.SEMarshalXML() 
Or simply wfutils.MarshalXML()